tools/ac.py

```diff
@@ -40,42 +71,70 @@ def prerun(self):
             if not ready[0]:
                 # print("Timeout connecting to AC server")
                 continue
-            _ = self.sock.recv(2048)
-            
+            #  get the car name to capture the appropriate max rpms
+            handshake_response = self.sock.recv(2048)
+            car_name = (
+                handshake_response[:50]
+                .decode("utf-8")
+                .strip("\x00")
+                .replace("\x00", "")
+            )[:-1]
+            # print("Car name:", car_name)
+            # breakpoint()
+            if self.revmax is None:
+                self.revmax = int(car_data[car_name])
+                print(car_name)
+                print("Max revs:", self.revmax)
+
             # confirm
             AcClient.client_data(self.sock, AcClient.SUBSCRIBE_UPDATE)
             break
 
     def postrun(self):
         AcClient.client_data(self.sock, AcClient.DISMISS)
+        self.revbase = None
+        self.revmax = None
+        self._gear = 0
         self.sock.setblocking(1)
 
     def tick(self):
-        ready = select.select([self.sock], [], [], .2)
+        # https://docs.google.com/document/d/1KfkZiIluXZ6mMhLWfDX1qAGbvhGRC3ZUzjVIt5FQpp4/pub
+
+        ready = select.select([self.sock], [], [], 0.2)
         if not ready[0]:
-            print('Timeout waiting for AC server data,', self.timeout_cnt)
+            print("Timeout waiting for AC server data,", self.timeout_cnt)
             self.timeout_cnt += 1
             if self.timeout_cnt > 10:
-                raise Exception('Too many timeouts received!')
+                raise Exception("Too many timeouts received!")
             return False
-        
-        # https://docs.google.com/document/d/1KfkZiIluXZ6mMhLWfDX1qAGbvhGRC3ZUzjVIt5FQpp4/pub
+
         self.timeout_cnt = 0
         data = self.sock.recv(2048)
-        self._speedKmh = int(struct.unpack('<f', data[8:12])[0])
+        self._speedKmh = int(struct.unpack("<f", data[8:12])[0])
         self._absInAction = bool(data[21])
         self._tcInAction = bool(data[22])
-        self._revLightsPercent = AcClient.rpms_to_revlights(int(struct.unpack('<f', data[68:72])[0]), 8000)
-        self._gear = int.from_bytes(data[76:80], byteorder='little') - 1
+
+        #  handling the rev measurement
+        if self.revbase is None:
+            time.sleep(0.5)  # wait for base rpms to stabilise
```

Comment:
This sleep has no effect, `data` doesn't change after fetching it from the socket ...

Reply:
fixed.